content/OTP/Guides/OTP_Walk-through.adoc

```diff
@@ -0,0 +1,101 @@
+
+Your authentication module defines the communications and actions between three components:
+
+* YubiKey – The user’s client device
+* Validation server – This can be either the YubiCloud or your own self-hosted validation service with the Internal YubiKey OTP Validation Service integrated.
+* Authentication Business Logic module – When one YubiKey provides access to multiple applications on a framework.
+
+
+image::placeholder-diagram.png[]
+**Figure 1 - Interaction between components**
+
+For additional considerations and best practices, see OTP Getting Started.
+
+=== Define Client Module Tasks
+
+The purpose of the client module is to:
+
+* Implement OTP transport protocol
+* Parse response from validation server (server-side)
+
+See https://developers.yubico.com/OTP/Libraries/Using_a_library.html[Using a Yubico OTP Library].
+
+*Step 1*: Create YubiCloud client, enable sending requests:
+
+....
+# clientId and secretKey is retrieved from https://upgrade.yubico.com/getapikey
+client = Yubico(clientId, secretKey)
+....
+
+*Step 2*: Verify OTPs submitted:
+....
+# otp is the OTP from the Yubikey
+otp_is_valid = client.verify(otp)
+....
+*Step 3*: Verify user login:
+....
+assert otp[:12] == user.yubikey_id
+....
+
+=== Define YubiKey Provisioning Method
+
+Provisioning is assigning a YubiKey ID to a User ID. At a minimum, administrators must be able to do this. Configure the database compatible with your applications to store the provisioning information: `YubiKey ID:User ID pairs.`
+
+Optionally, enable users to provision their own YubiKeys. Provide a website interface for two actions:
+
+1. User register, submit and verify OTP to validate YubiKey.
+2. User prompt for YubiKey touch, when YubiKey is validated.
+
+See https://developers.yubico.com/OTP/Libraries/Using_a_library.html[Using a Yubico OTP Library].
+
+Provision YubiKeys:
+....
+# Associate a user with a YubiKey ID
+user.yubikey_id = otp[:12]
+....
+
+
+=== Plug-ins for Creating your YubiKey OTP Module
+
+View and download the relevant plug-in component, which includes the Yubico Java client https://github.com/Yubico/yubico-java-client/tree/master/jaas[JAAS] and the https://github.com/Yubico/yubico-java-client/tree/master/demo-server[demo server].
+
+https://developers.yubico.com/yubico-pam/[Yubico PAM module] – Pluggable Authentication Modules (PAM) for GNU/Linux, Solaris and Mac OS X for user authentication.
+
+This requires the Yubico C client, https://developers.yubico.com/yubico-c-client/[libykclient], https://github.com/Yubico/yubico-pam[libpam-dev], cURL, https://developers.yubico.com/yubico-c/[libyubiky], and the YubiKey Personalization Tool, https://developers.yubico.com/yubikey-personalization/[yubikey-personalization]
+
+
+=== Libraries for Creating your YubiKey OTP Module
+
+View and download the relevant Yubico library component, https://developers.yubico.com/php-yubico/[PHP], which includes the PHP classes for Yubico authentication:
```

Comment:
Same comment here as previous section.

Reply:
Ditto. I don't see the point of this section, given that we already have https://developers.yubico.com/OTP/Libraries/. Please remove both this section and the Plugins section if you see fit.